compiler/crates/relay-lsp/src/rename/mod.rs

```diff
@@ -0,0 +1,638 @@
+                fragment_name: fragment_definition.name.value,
+                argument_name: variable_name,
+            };
+
+            Ok(RenameRequest::new(kind, variable_location))
+        }
+        ExecutableDefinition::Operation(operation_definition) => {
+            if !operation_definition
+                .variable_definitions
+                .as_ref()
+                .map_or(false, |defs| {
+                    defs.items.iter().any(|v| v.name.name == variable_name)
+                })
+            {
+                return Err(LSPRuntimeError::UnexpectedError(
+                    "Couldn't find variable definition for variable".into(),
+                ));
+            }
+
+            let operation_name = operation_definition
+                .name
+                .ok_or(LSPRuntimeError::ExpectedError)?
+                .value;
+            let kind = RenameKind::VariableDefinition {
+                variable_name,
+                operation_name,
+            };
+
+            Ok(RenameRequest::new(kind, variable_location))
+        }
+    }
+}
+
+fn map_locations_to_text_edits(
+    locations: Vec<IRLocation>,
+    new_text: String,
+    root_dir: &PathBuf,
+) -> HashMap<Url, Vec<TextEdit>> {
+    let vec_res: Vec<(Url, TextEdit)> = locations
+        .par_iter()
+        .flat_map(|location| {
+            let transformed = transform_relay_location_to_lsp_location(root_dir, *location);
+            transformed.ok().map(|lsp_location| {
+                let text_edit = TextEdit {
+                    range: lsp_location.range,
+                    new_text: new_text.to_owned(),
+                };
+                (lsp_location.uri, text_edit)
+            })
+        })
+        .collect();
+
+    let mut changes: HashMap<Url, Vec<TextEdit>> = HashMap::new();
+    for (uri, text_edit) in vec_res {
+        changes.entry(uri).or_default().push(text_edit);
+    }
+
+    changes
+}
+
+fn extract_parent_type(docblock: DocblockIr) -> StringKey {
+    match docblock {
+        DocblockIr::LegacyVerboseResolver(resolver_ir) => match resolver_ir.on {
+            On::Type(on_type) => on_type.value.item,
+            On::Interface(on_interface) => on_interface.value.item,
+        },
+        DocblockIr::TerseRelayResolver(resolver_ir) => resolver_ir.type_.item,
+        DocblockIr::StrongObjectResolver(strong_object) => strong_object.type_name.value,
+        DocblockIr::WeakObjectType(weak_type_ir) => weak_type_ir.type_name.value,
+    }
+}
+
+#[derive(Debug, Clone)]
+struct FragmentFinder {
+    fragment_locations: Vec<IRLocation>,
+    fragment_name: StringKey,
+}
+
+impl FragmentFinder {
+    pub fn get_fragment_usages(program: &Program, name: StringKey) -> Vec<IRLocation> {
+        let mut finder = FragmentFinder {
+            fragment_locations: vec![],
+            fragment_name: name,
+        };
+        finder.visit_program(program);
+        finder.fragment_locations
```

Comment:
Fragment names can also be referenced in Relay resolver docblocks in the `@rootFragment` tag. Would be cool if we could handle those as well.

Reply:
I've been looking into this and I think I'll have to introduce tracking of those `@rootFragment` references (with location information) within the `Program`. Currently the `@rootFragment` name is only tracked as argument on the internal relay resolver field directive. I don't see how I could include location information there, without weirdly serializing it to GraphQL primitives.
If you have a nice idea on how to tackle this - let me know :)